ll-token-field.html

```diff
@@ -110,18 +167,32 @@
         e.stopPropagation();
         var newTag = input.value;
         input.value = '';
-        this.tags = this.tags.concat(newTag);
+        this.push('tags', newTag);
+        this.tagChanged();
         input.focus();
       }
       else if (e.keyCode === 8 && input.value === '') { // Backspace
-        this.tags = this.tags.slice(0, -1);
+        this.splice('tags', (this.tags.length -1), 1);
+        this.tagChanged();
       }
     },
 
     _removeToken: function(e) {
-      var removed = e.model.arrayDelete(this.tags, e.model.item);
-      this.tags = _.without(this.tags, removed);
+      this.splice('tags', e.model.index, 1 );
+      this.tagChanged();
+      e.preventDefault();
+      e.stopPropagation();
     },
+
+    _focusInput: function() {
+      var input = this.$$('#tag-input');
+      input.focus();
+    },
+    tagChanged: function() {
+      //needing to hard reset the tags object to get it reflected to the attribute
+      this.tags = this.tags.splice(0);
+      this.fire('ll-token-modified', this.tags)
```

Comment:
I added a test for the removal of the tokens, it ensures that the event is firing with the proper data passed. I also have tests for this on the ll-property-image (for both adding and removing tags to ensure I get the events)